a10_octavia/controller/worker/controller_worker.py

```diff
@@ -1502,3 +1502,45 @@ def _get_flavor_data(self, flavor_id):
             flavor_data = json.loads(flavor_profile.flavor_data)
             return flavor_data
         return None
+
+    def delete_load_balancer_with_housekeeping(self, pending_lb, cascade=True):
+        """Function to delete load balancer for A10 provider using Housekeeper thread"""
+        if pending_lb.project_id in CONF.hardware_thunder.devices:
+            try:
+                self._lb_repo.update(db_apis.get_session(),
+                                     pending_lb.id,
+                                     provisioning_status=constants.ERROR)
+                vthunder = self._vthunder_repo.get_vthunder_from_lb(
+                    db_apis.get_session(), pending_lb.id)
+                if vthunder is not None:
```

Comment:
if vthunder.compute_id is not None, we should skip the LB.
This is rare but may happen when user restart a10-octavia with different configuration.

Reply:
done